sdk/core/azure-core/azure/core/pipeline/_tools.py

```diff
@@ -46,21 +46,20 @@ def to_rest_request(pipeline_transport_request):
 def to_rest_response(pipeline_transport_response):
     from .transport._requests_basic import RequestsTransportResponse
     from ..rest._requests_basic import RestRequestsTransportResponse
-    from ..rest import HttpResponse
     if isinstance(pipeline_transport_response, RequestsTransportResponse):
         response_type = RestRequestsTransportResponse
     else:
-        response_type = HttpResponse
+        raise ValueError("Unknown transport response")
```

Comment:
Is this relying on the fact that we only have this sync transport today?

Reply:
Yeah. And now that `azure.core.rest.HttpResponse` is an ABC, can't return it as a fallback

Reply:
We can provide a default concrete fallback implementation, though.